packages/protocol/src/peer/InteractionQueue.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @license
+ * Copyright 2022-2025 Matter.js Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Environment, Environmental, PromiseQueue } from "@matter/general";
+
+const CONCURRENT_QUEUED_INTERACTIONS = 4;
+const INTERACTION_QUEUE_DELAY_MS = 100;
+
+export class InteractionQueue extends PromiseQueue {
```

Comment:
maybe better name ClientInteractionQueue? maybe we somewhen also need one for server

Reply:
It's in a directory called "peer" so is already associated logically with client implementation, but there's not really anything client-specific about the implementation so I'd tend to leave the name more generic until either there's a conflict, or move to a more general place if it becomes more generally applicable

Reply:
kk